shotgun_api3/shotgun.py

```diff
@@ -2136,7 +2126,7 @@ def schema_field_update(self, entity_type, field_name, properties, project_entit
             "field_name": field_name,
             "properties": [
                 {"property_name": k, "value": v}
-                for k, v in six.iteritems((properties or {}))
+                for k, v in (properties.items() if isinstance(properties, dict)  else {})
```

Comment:
I don't think this is correct. What you want is:
```suggestion
                for k, v in ((properties if isinstance(properties, dict) else {}).items())
```

But I think we'd better have this check done earlier in the function instead.